include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+            return;
+        }
+        event.failure = ev.at("value").at("failure");
+        router->on_status_end(event);
+        return;
+    }
+    if (ev.at("key") == events::StatusGeoipLookupEvent::key) {
+        events::StatusGeoipLookupEvent event;
+        if (ev.at("value").count("probe_ip") <= 0 || !ev.at("value").at("probe_ip").is_string()) {
+            // TODO(bassosimone): route this error.
+            interrupt();
+            return;
+        }
+        event.probe_ip = ev.at("value").at("probe_ip");
+        if (ev.at("value").count("probe_asn") <= 0 || !ev.at("value").at("probe_asn").is_string()) {
+            // TODO(bassosimone): route this error.
+            interrupt();
+            return;
+        }
+        event.probe_asn = ev.at("value").at("probe_asn");
+        if (ev.at("value").count("probe_cc") <= 0 || !ev.at("value").at("probe_cc").is_string()) {
+            // TODO(bassosimone): route this error.
+            interrupt();
+            return;
+        }
+        event.probe_cc = ev.at("value").at("probe_cc");
+        if (ev.at("value").count("probe_network_name") <= 0 || !ev.at("value").at("probe_network_name").is_string()) {
+            // TODO(bassosimone): route this error.
+            interrupt();
+            return;
+        }
+        event.probe_network_name = ev.at("value").at("probe_network_name");
+        router->on_status_geoip_lookup(event);
+        return;
+    }
+    if (ev.at("key") == events::StatusProgressEvent::key) {
+        events::StatusProgressEvent event;
+        if (ev.at("value").count("percentage") <= 0 || !ev.at("value").at("percentage").is_number_float()) {
+            // TODO(bassosimone): route this error.
+            interrupt();
+            return;
+        }
+        event.percentage = ev.at("value").at("percentage");
+        if (ev.at("value").count("message") <= 0 || !ev.at("value").at("message").is_string()) {
+            // TODO(bassosimone): route this error.
+            interrupt();
+            return;
+        }
+        event.message = ev.at("value").at("message");
+        router->on_status_progress(event);
+        return;
+    }
+    if (ev.at("key") == events::StatusQueuedEvent::key) {
+        events::StatusQueuedEvent event;
+        /* No attributes */
+        router->on_status_queued(event);
+        return;
+    }
+    if (ev.at("key") == events::StatusMeasurementStartEvent::key) {
+        events::StatusMeasurementStartEvent event;
+        if (ev.at("value").count("idx") <= 0 || !ev.at("value").at("idx").is_number_integer()) {
+            // TODO(bassosimone): route this error.
+            interrupt();
+            return;
+        }
+        event.idx = ev.at("value").at("idx");
+        if (ev.at("value").count("input") <= 0 || !ev.at("value").at("input").is_string()) {
+            // TODO(bassosimone): route this error.
+            interrupt();
+            return;
+        }
+        event.input = ev.at("value").a
```

Comment:
This part is of course not autogenerated. It interrupts the test if we see an unmapped event.